### PR TITLE
Harden GC WAL error handling

### PR DESCRIPTION
**Pull request by yacovm**
- Properly handle error return values where the error is ignored
- Properly increment the cumulative size of the WAL when appending records.
- Add unit tests for edge cases

### CODE REVIEW

File: wal/gc.go [hunk 1]
```diff
@@ -125,16 +127,17 @@ func (gcw *GarbageCollectedWAL) Append(payload []byte) error {
 			retentionTerm: rt,
 			size:          len(payload),
 		})
+	} else {
+		// We're still using the last WAL, so we need to update its size
+		gcw.wals[len(gcw.wals)-1].size += len(payload)
```
Comment (samliok) [inline wal/gc.go:132]: surprised we never caught this

**Review by samliok — APPROVED** [review]

File: wal/gc.go (hunk 1 above)
Comment (yacovm) [reply wal/gc.go:132]: because we never used the GC WAL until now
